query-engine/core/src/query_graph_builder/write/upsert.rs

```diff
@@ -238,6 +240,7 @@ fn can_use_connector_native_upsert(
         && !empty_update
         && !has_nested_selects
         && where_values_same_as_create
+        && !connector_ctx.relation_mode.is_prisma()
```

Comment:
I am not clear on this line. Do native upserts interfere with the referential actions emulation?

Reply:
It does interfere, unfortunately. The emulated referential actions only need to be executed when doing an update, but we lose that information when running a native upsert. It's a single operation that "opaquely" does one or the other. There might be a way to make this work, but I need to investigate what native upsert returns for each database we support. In any case, this shouldn't affect 99% of users atm. We only support native upsert on Postgres, Cockroachdb and Sqlite. Those aren't connectors that are likely to be used with `relationMode = "prisma"` because they have a built-in referential integrity system.

Until someone complains, I thought it'd be best to simply ignore native upsert when `relationMode = prisma`.

Reply:
I'm OK with that :+1: